src/cfnlint/rules/resources/rds/AuroraScalingConfiguration.py

```diff
@@ -0,0 +1,41 @@
+"""
+Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+from cfnlint.rules import CloudFormationLintRule
+from cfnlint.rules import RuleMatch
+
+
+class AttributeMismatch(CloudFormationLintRule):
+    """ScalingConfiguration only set for Aurora Serverless"""
+    id = 'E3028'
+    shortdesc = 'ScalingConfiguration only set for Aurora Serverless'
+    description = 'You cannot specify ScalingConfiguration for non Aurora Serverless AWS::RDS::DBCluster'
+    source_url = 'https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#cfn-rds-dbcluster-scalingconfiguration'
+    tags = ['resources', 'rds']
+
+    def __init__(self):
+        """Init"""
+        super(AttributeMismatch, self).__init__()
+        self.resource_property_types = ['AWS::RDS::DBCluster']
+
+    def check(self, properties, path, cfn):
+        """Check itself"""
+        matches = []
+        property_sets = cfn.get_object_without_conditions(properties)
```

Comment:
Can we change this to?
`property_sets = cfn.get_object_without_conditions(properties, ['EngineMode', 'ScalingConfiguration'])`

This will prevent some performance issues where people over condition all the properties inside this resource type.